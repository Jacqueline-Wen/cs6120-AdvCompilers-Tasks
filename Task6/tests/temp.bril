@main(m: bool) {
  a: int = const 3;
  br m .left .right;
.left:
  a: int = add a a;
  br m .end1 .end2;
.right:
  a: int = mul a a;
  br m .end1 .end2;
.end1:
  print a;
.end2:
  print a;
}