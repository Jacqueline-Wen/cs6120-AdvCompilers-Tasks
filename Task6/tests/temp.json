{
  "functions": [
    {
      "args": [
        {
          "name": "m",
          "type": "bool"
        }
      ],
      "instrs": [
        {
          "dest": "a",
          "op": "const",
          "type": "int",
          "value": 3
        },
        {
          "args": [
            "m"
          ],
          "labels": [
            "left",
            "right"
          ],
          "op": "br"
        },
        {
          "label": "left"
        },
        {
          "args": [
            "a",
            "a"
          ],
          "dest": "a",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "m"
          ],
          "labels": [
            "end1",
            "end2"
          ],
          "op": "br"
        },
        {
          "label": "right"
        },
        {
          "args": [
            "a",
            "a"
          ],
          "dest": "a",
          "op": "mul",
          "type": "int"
        },
        {
          "args": [
            "m"
          ],
          "labels": [
            "end1",
            "end2"
          ],
          "op": "br"
        },
        {
          "label": "end1"
        },
        {
          "args": [
            "a"
          ],
          "op": "print"
        },
        {
          "label": "end2"
        },
        {
          "args": [
            "a"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
