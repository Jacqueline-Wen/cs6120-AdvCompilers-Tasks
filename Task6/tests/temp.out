[
  {
    "dest": "m_2",
    "op": "arg",
    "type": "bool"
  },
  {
    "dest": "a_3",
    "op": "const",
    "type": "int",
    "value": 3
  },
  {
    "args": [
      "m_2"
    ],
    "labels": [
      "left",
      "right"
    ],
    "op": "br"
  },
  {
    "args": [
      "a_3",
      "a_3"
    ],
    "dest": "a_4",
    "op": "add",
    "type": "int"
  },
  {
    "args": [
      "m_2"
    ],
    "labels": [
      "end1",
      "end2"
    ],
    "op": "br"
  },
  {
    "args": [
      "a_3",
      "a_3"
    ],
    "dest": "a_5",
    "op": "mul",
    "type": "int"
  },
  {
    "args": [
      "m_2"
    ],
    "labels": [
      "end1",
      "end2"
    ],
    "op": "br"
  },
  {
    "args": [
      "a_4",
      "a_5"
    ],
    "dest": "a_0",
    "labels": [
      2,
      3
    ],
    "op": "phi"
  },
  {
    "args": [
      "a_0"
    ],
    "op": "print"
  },
  {
    "args": [
      "a_4",
      "a_5",
      "a_0"
    ],
    "dest": "a_1",
    "labels": [
      2,
      3,
      4
    ],
    "op": "phi"
  },
  {
    "args": [
      "a_1"
    ],
    "op": "print"
  }
]
[
  {
    "dest": "m_2",
    "op": "arg",
    "type": "bool"
  },
  {
    "dest": "a_3",
    "op": "const",
    "type": "int",
    "value": 3
  },
  {
    "args": [
      "m_2"
    ],
    "labels": [
      "left",
      "right"
    ],
    "op": "br"
  },
  {
    "args": [
      "a_3",
      "a_3"
    ],
    "dest": "a_4",
    "op": "add",
    "type": "int"
  },
  {
    "args": [
      "a_4"
    ],
    "dest": "a_0",
    "op": "id"
  },
  {
    "args": [
      "a_4"
    ],
    "dest": "a_1",
    "op": "id"
  },
  {
    "args": [
      "m_2"
    ],
    "labels": [
      "end1",
      "end2"
    ],
    "op": "br"
  },
  {
    "args": [
      "a_3",
      "a_3"
    ],
    "dest": "a_5",
    "op": "mul",
    "type": "int"
  },
  {
    "args": [
      "a_5"
    ],
    "dest": "a_0",
    "op": "id"
  },
  {
    "args": [
      "a_5"
    ],
    "dest": "a_1",
    "op": "id"
  },
  {
    "args": [
      "m_2"
    ],
    "labels": [
      "end1",
      "end2"
    ],
    "op": "br"
  },
  {
    "args": [
      "a_0"
    ],
    "op": "print"
  },
  {
    "args": [
      "a_0"
    ],
    "dest": "a_1",
    "op": "id"
  },
  {
    "args": [
      "a_1"
    ],
    "op": "print"
  }
]
